app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -366,6 +367,15 @@ class ComponentsValidator(
     }
   }
 
+  private def checkEORI(fieldValue: FormComponent, value: String) = {
+    val ValidCRN = "^[A-Z]{2}[0-9]{12}|^[A-Z]{2}[0-9]{15}$".r
```

Comment:
I wonder if the regex needs two ^ but only one $ , I have suggested extra tests below to help